cli/xml-to-ts.ts

```diff
@@ -0,0 +1,236 @@
+import * as fs from "fs";
+import * as path from "path";
+import * as xmldom from "xmldom";
+const xpath = require("xpath");
+
+/**
+ * iface describes a TypeScript interface
+ */
+interface iface {
+  name: string;
+  fields: field[];
+}
+
+/**
+ * field describes an object property
+ */
+interface field {
+  name: string;
+  type: string;
+  comment?: string;
+  optional: boolean;
+
+  subfields?: field[];
+}
+
+/**
+ * ifaceToTs formats an iface as a TypeScript interface
+ * @param iface
+ */
+function ifaceToTs(iface: iface): string {
+  return `export interface ${iface.name} {
+${iface.fields.map(f => fieldfmt(f)).join("\n")}
+};
+`;
+}
+
+function fieldfmt(f: field, padding: string = "  "): string {
+  const optional = f.optional ? "?" : "";
+  const comment = f.comment ? ` // ${f.comment}` : "";
+  const subfields =
+    f.subfields && f.subfields.length > 0
+      ? `<{
+${f.subfields.map(f => fieldfmt(f, padding + "  ")).join("\n")}
+${padding}}>`
+      : "";
+  return `${padding}${f.name}${optional}: ${f.type}${subfields}${comment}`;
+}
+
+/**
+ * xmlToIface parses an xml-file and returns an iface
+ * @param name the name of the interface
+ * @param xml
+ */
+function xmlToIface(name: string, xml: string): iface {
+  const iface: iface = {
+    name,
+    fields: []
+  };
+  const doc = new xmldom.DOMParser().parseFromString(xml);
+  const form = xpath
+    .evaluate("//form", doc, null, xpath.XPathResult.ANY_TYPE, null)
```

Comment:
Maybe it would be nicer to create a wrapper function (or extension function) around `evaluate`, so we don't have to write all these ugly `nulls` all over the place.